### PR TITLE
Add tests publish bad arguments

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Comment by Lobotuerk** [comment]
Tests failing as (I think) implementations are not returning whats expected according to API documents

File: test_rmw_implementation/test/test_publisher.cpp
Comment (hidmic) [inline test_rmw_implementation/test/test_publisher.cpp:381]: @Lobotuerk it'd seem like you're missing a `test_msgs__msg__BasicTypes__fini` call to not leak the message once the test is finished.

File: test_rmw_implementation/test/test_publisher.cpp
Comment (hidmic) [inline test_rmw_implementation/test/test_publisher.cpp:385]: @Lobotuerk why not using `&input_message` directly elsewhere?

File: test_rmw_implementation/test/test_publisher.cpp
Comment (hidmic) [inline test_rmw_implementation/test/test_publisher.cpp:379]: @Lobotuerk nit: consider 
```suggestion
TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), publish_message_with_bad_arguments) {
```

**Review by hidmic — COMMENTED** [review]
@Lobotuerk tests are pretty simple, I'd suspect implementations indeed.

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Blast545) [inline test_rmw_implementation/test/test_publisher.cpp:382]: Remove not needed comments before merging

**Review by Blast545 — COMMENTED** [review]
LGTM after addressing pending comments + green CI

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:381]: I was seeing other tests that did not have the finish after initializing the basic messages

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:385]: Makes sense

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:379]: Uncrustify does not seem to like that way, I'm changing it to 
```
TEST_F(
  CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION),
  publish_message_bad_arguments) {
```

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:382]: Removed

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:381]: Anyways, added

Comment (Blast545) [reply test_rmw_implementation/test/test_publisher.cpp:383]: If you have any references to tests that have success `ini` calls without `fini` calls without you can raise an issue or open a PR, because they are probably leaking memory and is not desired.

**Review by Blast545 — APPROVED** [review]
LGTM pending green CI

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:383]: @Blast545 https://github.com/ros2/rmw_implementation/blob/master/test_rmw_implementation/test/test_serialize_deserialize.cpp,
 maybe I'm wrong, but I see 3 inits and no fini

Comment (hidmic) [reply test_rmw_implementation/test/test_publisher.cpp:383]: That's an omission of my own! We have to fix it :)

File: test_rmw_implementation/test/test_publisher.cpp
Comment (hidmic) [reply test_rmw_implementation/test/test_publisher.cpp]: Perhaps `publish_with_bad_arguments`?

File: test_rmw_implementation/test/test_publisher.cpp
Comment (hidmic) [inline test_rmw_implementation/test/test_publisher.cpp:384]: @Lobotuerk nit^N: 
```suggestion
  rmw_publisher_allocation_t * null_allocation{nullptr};  // still valid
```

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp]: Changed

File: test_rmw_implementation/test/test_publisher.cpp
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp]: Changed

**Comment by hidmic** [comment]
@ros-pull-request-builder retest this please.

File: test_rmw_implementation/test/test_publisher.cpp
Comment (hidmic) [inline test_rmw_implementation/test/test_publisher.cpp:386]: @Lobotuerk `default_allocation` -> `null_allocation`.

Same elsewhere.

Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp]: Fixed on ed87738

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12394/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7362/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10105/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12307/)

**Comment by Lobotuerk** [comment]
CI jobs after master rebase:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12426/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7393/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10136/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12339/)

**Comment by Lobotuerk** [comment]
Ci green, merging

**Pull request merged**
